### ISSUE TITLE
Wan 2.1 14B T2V LoRA Training Issue

### ISSUE THREAD

**Issue by AILova** [tdrussell/diffusion-pipe#128]
Anyone have luck training a LoRA for the 14B T2V Wan 2.1 model for use in ComfyUI?   Quick background - I was able to successfully create and use a LoRA for 1.3B T2V model, but when I try to create a LoRA using the 14B T2V model, diffusion pipe creates the LoRA fine (it's a 299MB safetensor file), however, when I try to use it in ComfyUI it just has no effect on the resulting video.  Further, ComfyUI throws no error nor does the log file have any errors.  But it just appears to ignore the LoRA.   Also, I'm a little confused because the official Wan 2.1 14B T2V repo I used to train the LoRA shows 6 diffusion-pytorch models totaling ~53GB, whereas the Wan 2.1 14B T2V safetensors file used in ComfyUI is just ~14GB.  Seems like maybe I'm mixing things up incorrectly.  

Bottomline - can of you smarter people let me know if it's possible to train a LoRA for Wan 2.1 T2V 14B model for use in ComfyUI and run both the training and inference on a 24GB 4090 RTX?  And if so, how ? :) 

Thanks in advance.

**Comment by tdrussell** [comment]
Are you using torch.compile nodes with the ComfyUI native implementation? These will (silently?) ignore any lora. With Kijai's wrapper custom nodes torch.compile will work with lora. There is also a node in the KJNodes pack that changes the model patcher order, so that loras can work with torch compile even with the native implementation.

**Comment by AILova** [comment]
thanks for the quick response, but no, I'm using a very basic workflow with no speed-ups (no teacache, no torch compile, etc)... its just basic native Comfyui nodes (i.e., I didn't use any of the Kijai Wan nodes.).  Could try using Kijai's nodes and model files but I don't  think that is the issue given the 1.3B LoRA I made works fine on the same basic workflow (granted when I load the corresponding 1.3B wan model).  Isn't something wrong considering the Wan 14B T2V repo is ~53GB for the diffusion model, whereas the ComfyUI Wan 2.1 14B T2V model is just 14GB, or is this normal?  I wonder if anyone has had success doing this because I find it odd that no LoRAs for 14GB Wan have yet to appear on Civitai, etc.  Any help is appreciated and thanks for your diffusion pipe software, I've had amazing results using it for Hunyuan and others!

**Comment by tdrussell** [comment]
The ComfyUI file is just the transformer part of the model, it should be 14GB in fp8. I am sure that lora training works for 14b, you can find several loras on civit that are most definitely having an effect.

Have you looked at the epoch-level loss in tensorboard? You can go to the output directory, run "tensorboard --logdir=.", then open the link. The epoch loss graph is averaging the loss over each epoch, even then it is noisy, but it should generally go down over time. You can also split off a validation data set, that loss is stabilized with fixed RNG seeds and should show a smooth decreasing curve. These are ways you can confirm the lora is learning something.

**Comment by AILova** [comment]
Ok, thanks, I will give those a shot.

**Comment by kanarch66** [comment]
any luck @AILova  ?

**Comment by AILova** [comment]
Thanks for reaching back out. Yes, I think my initial issue was that I didn't set dtype to 'bfloat16' under the adapter section.  If one doesn't do so (which is not uncommented by default in the Wan template) then it will train but it will have no effect on the sampling.  It now works great.  Thx.
